astropy/coordinates/baseframe.py

```diff
@@ -236,17 +236,17 @@ class BaseCoordinateFrame(ShapedLikeNDArray, metaclass=FrameMeta):
 
     Parameters
     ----------
-    representation : `BaseRepresentation` or None
-        A representation object or `None` to have no data (or use the other
+    representation_type : `BaseRepresentation` or None
+        A representation class or ``None`` to have no data (or use the other
         arguments)
     *args, **kwargs
-        Coordinates, with names that depend on the subclass.
-    differential_cls : `BaseDifferential`, dict, optional
+        Coordinate components, with names that depend on the subclass.
+    differential_type : `BaseDifferential`, dict, optional
```

Comment:
And above you need to add `representation_type` (as in the initializer below).

I also note that the line is confusing: it should be 
```
differential_type: `BaseDifferential` subclass, str, dict, optional
```